app/services/users/profile.rb

```diff
@@ -63,12 +64,21 @@ def account
       nil
     end
 
+    def demographics
+      demographic_svc = VAProfile::Demographics::Service.new @user
+      demographic_svc.get_demographics&.demographics
+    rescue => e
+      scaffold.errors << Users::ExceptionHandler.new(e, 'VAProfile').serialize_error
+      nil
+    end
+
     def profile
       {
         email: user.email,
         first_name: user.first_name,
         middle_name: user.middle_name,
         last_name: user.last_name,
+        preferred_name: demographics&.preferred_name&.text,
```

Comment:
Chaining safe navigation operators like this is generally not a good practice and can be a code smell. It can make things harder to read or understand where the nil values are coming from.

For instance, if this value were to be blank, does that mean...
* the call to the demographics API failed?
* the user doesn't have a preferred name?
* the value for preferred name doesn't respond to the `text` message being sent?

I looked at the underlying service and the associated response classes, and I'm not too certain there's much you can do here to address my concern.

All that being said, I think this particular value should be added to the `mpi_profile` return value instead. [This line](https://github.com/department-of-veterans-affairs/vets-api/blob/0253f4695a40f9640e1a3bc41a9d8d1b4c3e754f/app/services/users/profile.rb#L52) seems to extract all of the VA profile data from that method.

Reply:
@dickdavis Thank you for the feedback.
1. I think I can wrap some of that logic into my new function and properly log a failure. I will try that out.
2. The problem I have with adding that value to the `mpi_profile` is that it has nothing to do with VA Profile even when the name of the attribute in the scaffold is called `va_profile`. All the values in that section come from MPI, and `mpi_profile` even has an error catcher for the MPI service. This is the reason I placed the preferred name in the `profile` section. I can still just add `preferred name` to `mpi_profile` as my new function will catch the error properly, but I just feel we are mixing sources there. Of course, I am saying all this with no real history on what was intended for each of those functions. Does it make sense to keep that separate, or I am reading too much into how the code should work?

Reply:
Ha, I think we are probably both in the same boat with regard to knowing the original intention.

I think I have an idea. What if we tried...

* Add `demographics` to the scaffold [here](https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/services/users/scaffold.rb#L11)
```ruby
class Scaffold < Struct.new(... :demographics, ...)
```

* Update the `fetch_and_serialize_profile` method [here](https://github.com/department-of-veterans-affairs/vets-api/blob/ea8f21d88045970d5f8830ec96bf5577efcddbcc/app/services/users/profile.rb#L48) to scaffold an entirely new spot for demographics. This should make it cleaner to add additional data pulled through calls using the demographics service in the future. Something like this:
```ruby
    def fetch_and_serialize_profile
      ...
      scaffold.demographics = demographics_data
      ...
    end
```

* Then, implement the `demographics_data` method (name it whatever you like really):
```ruby
    def demographics_data
      demographic_svc = VAProfile::Demographics::Service.new @user
      # We want to catch if preferred_name or text is missing, so no safe operator here.
      data = demographic_svc.get_demographics.demographics
      {
        preferred_name: data.preferred_name.text
      }
    rescue => e
      scaffold.errors << Users::ExceptionHandler.new(e, 'VAProfile').serialize_error
      nil
    end
```

* Add the `demographics` attribute to the `UserSerializer` [here](https://github.com/department-of-veterans-affairs/vets-api/blob/ea8f21d88045970d5f8830ec96bf5577efcddbcc/app/serializers/user_serializer.rb#L9) and delegate the method call to the object somewhere around [here](https://github.com/department-of-veterans-affairs/vets-api/blob/ea8f21d88045970d5f8830ec96bf5577efcddbcc/app/serializers/user_serializer.rb#L25)
```ruby
  attributes ... :demographics, ...

  ...

  delegate :demographics, to: :object
```